### PR TITLE
Directional computer sprites

### PR DESCRIPTION
**Pull request by ike709**
:cl: ike709
imageadd: Added directional computer sprites. Maps haven't been changed yet.
/:cl:

Stolen from FTL13. Not sure where FTL13 stole them from.

https://i.gyazo.com/f0ce59140d20af815e803e4fd6018cd0.mp4
Arcade machine not included.

I apologize to the mappers.

### CODE REVIEW

**Comment by Qustinnus** [comment]
we took them from cev eris and Stobarico made any missing screens AFAIK.

**Review by ghost — CHANGES_REQUESTED** [review]
some consoles reset their direction so go fix that first loser :^)

**Comment by Cyberboss** [comment]
@lzimann @icondiffbot failed us

**Comment by ike709** [comment]
I need to fix DNA scanners changing the dir of their computers.

**Comment by PKPenguin321** [comment]
North facing sprite is a bit wonky

**Comment by ike709** [comment]
I say it looks pretty damn good, all things considered.

Now we just need directional vending machines.

**Comment by Qustinnus** [comment]
@ike709 You could potentially (I'm not sure if it'll work) fix the issue better by replacing "dir in list" with for example "direction in list"

by doing dir in list I'm guessing you're changing the value of the actual dir 4 times.

**Comment by ike709** [comment]
\>Floyd trying to improve his github cred by mentioning the fix he explained in Discord.

**Comment by Qustinnus** [comment]
i explained it here first and then called you a troglodyte on discord when you said you fixed it and it was easy

**Comment by ike709** [comment]
Technically I did fix it.

**Comment by Qustinnus** [comment]
technically gooncode functions

**Comment by ike709** [comment]
you won't function much longer if you don't hush

**Comment by calzilla1** [comment]
You know me, coe goes moo, ducks go quack, and calzilla says the new sprites look like shit

**Comment by praisenarsie** [comment]
@WJohn

**Comment by Dorsisdwarf** [comment]
It looks good, but I'm not quite sure I like the heavily blued metal tones of the consoles. What does it look like if you desaturate that a bit to make it closer to normal grey?

**Comment by WJohn** [comment]
I am... actually totally cool with these. I have a grey ramp of my own I can run these through if people insist they need changing but I'd be fine with these as-is, tbh.

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -54,8 +54,8 @@
 
 /obj/machinery/computer/scan_consolenew/Initialize()
 	. = ..()
-	for(dir in list(NORTH,EAST,SOUTH,WEST))
-		connected = locate(/obj/machinery/dna_scannernew, get_step(src, dir))
+	for(var/direction in list(NORTH,EAST,SOUTH,WEST))
```
Comment (lzimann) [inline code/game/machinery/computer/dna_console.dm]: use `GLOB.cardinals` instead of that list

File: code/game/machinery/dna_scanner.dm [hunk 2]
```diff
@@ -84,8 +84,8 @@
 		open_machine()
 
 /obj/machinery/dna_scannernew/proc/locate_computer(type_)
-	for(dir in list(NORTH,EAST,SOUTH,WEST))
-		var/C = locate(type_, get_step(src, dir))
+	for(var/direction in list(NORTH,EAST,SOUTH,WEST))
```
Comment (lzimann) [inline code/game/machinery/dna_scanner.dm]: use `GLOB.cardinals` instead of that list

File: code/game/machinery/computer/cloning.dm [hunk 3]
```diff
@@ -98,10 +98,9 @@
 /obj/machinery/computer/cloning/proc/findcloner()
 	var/obj/machinery/clonepod/podf = null
 
-	for(dir in list(NORTH,EAST,SOUTH,WEST))
-
-		podf = locate(/obj/machinery/clonepod, get_step(src, dir))
+	for(var/direction in list(NORTH,EAST,SOUTH,WEST))
```
Comment (lzimann) [inline code/game/machinery/computer/cloning.dm]: use `GLOB.cardinals` instead of that list

File: code/game/machinery/computer/cloning.dm [hunk 4]
```diff
@@ -83,10 +83,10 @@
 	var/obj/machinery/dna_scannernew/scannerf = null
 
 	// Loop through every direction
-	for(dir in list(NORTH,EAST,SOUTH,WEST))
+	for(var/direction in list(NORTH,EAST,SOUTH,WEST))
```
Comment (lzimann) [inline code/game/machinery/computer/cloning.dm]: use `GLOB.cardinals` instead of that list

**Comment by ike709** [comment]
@praisenarsie This doesn't make the maps unplayable. Computers have always faced south. This doesn't break them in any way. Mappers will just have the ability to give them a dir other than south.

Also, I assume mapping is against the freeze due to the circuitry lab and other stuff. I'm willing to alter the maps, but it'll have to wait, and I'll probably do each map in a separate PR anyways to reduce conflicts.

**Comment by WJohn** [comment]
Yeah it'll be kind of a pain in the ass to change all these manually, but it's not the worst thing ever to have a bunch of them still facing south.

**Pull request merged**